nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -659,9 +659,21 @@ Costmap2DROS::dynamicParametersCallback(std::vector<rclcpp::Parameter> parameter
       if (name == "width") {
         resize_map = true;
         map_width_meters_ = parameter.as_int();
+        if (map_width_meters_ <= 0) {
```

Comment:
I think these should be rejected -- e.g. not changed and the `result` is returned as false indicating the parameter requested change is invalid and rejected. You can elevate the log to an error message and remove setting default to 5 stuff since we'll just reject it.

Reply:
You are right! I've changed the logic as: if the value of the request is not sensible, just give an error message and do nothing else; if it's sensible ,run as before.